app/domain/authentication/authn_k8s/inject_client_cert.rb

```diff
@@ -157,6 +151,23 @@ def cert_to_install
       def k8s_object_lookup
         @k8s_object_lookup ||= K8sObjectLookup.new(webservice)
       end
+
+      # This code is implemented similarly also in application_identity.rb
+      # We have it here too as we need the container name for the injection
+      # and it simplifies the code to have this specific 2 methods duplicated
+      # rather than passing around the ApplicationIdentity object
+      def container_name
+        container_annotation_value("authn-k8s/#{@service_id}") ||
+          container_annotation_value("authn-k8s") ||
+          container_annotation_value("kubernetes") ||
```

Comment:
what is the difference here from 
```
          container_annotation_value("authn-k8s") ||
          container_annotation_value("kubernetes") ||
```
I know you are referencing the application_identity.rb file but please explain to me why we also need `container_annotation_value("kubernetes")` (:

Reply:
it is for backwards compatibility as before this PR the container name was in the `kubernetes/authentication-container-name` annotation.